vertx-mongo-client/src/main/java/io/vertx/ext/mongo/impl/MongoClientImpl.java

```diff
@@ -200,6 +207,25 @@ public void close() {
   }
 
   @Override
+  public io.vertx.ext.mongo.MongoClient findBatchWithOptions(String collection, JsonObject query, FindOptions options, Handler<AsyncResult<JsonObject>> resultHandler) {
+    requireNonNull(collection, "collection cannot be null");
+    requireNonNull(query, "query cannot be null");
+    requireNonNull(resultHandler, "resultHandler cannot be null");
+
+    FindIterable<JsonObject> view = doFind(collection, query, options);
+    Block<JsonObject> documentBlock = document -> wrapCallback(resultHandler).onResult(document, null);
+    SingleResultCallback<Void> callbackWhenFinished = (result, t) -> {
+      if (t != null) {
+        resultHandler.handle(Future.failedFuture(t));
```

Comment:
I'd personally prefer to see t get called throwable (what is a t? :-)), but if it's matching conventions in this class then leave it.